### ISSUE TITLE
Shap value doesn't match model output

### ISSUE THREAD

**Issue by didericksen** [shap/shap#669]
First off, thanks so much for shap! I love it. So I'm having an issue where the shap values are on a much different scale than my model output. I only come across this issue when I run lightgbm in random forest mode - when I use lightgbm in boosting mode or use sklearn random forest, shap works as expected. Here is a minimal reproducible example of my error.

code:
```
import lightgbm as lgb
import shap

# load JS visualization code to notebook
shap.initjs()

# load data
X, y = shap.datasets.boston()

# train lightgbm random forest
model = lgb.LGBMRegressor(boosting_type = 'rf', 
                          n_estimators = 500,
                          max_depth = 7,
                          num_leaves = 2**7,
                          subsample_freq = 1,
                          subsample = 0.6,
                          colsample_bytree = 0.5,
                          n_jobs = -1,
                          random_state = 123,
                          importance_type = 'gain',
                         )

model.fit(X, y)

# calculate shap values
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(X)

# visualize the first prediction's explanation
shap.force_plot(explainer.expected_value, shap_values[0,:], X.iloc[0,:])

# get the first prediction from the model
model.predict(X.iloc[0:1])
```
output:

visualizer's prediction is 14340.97
the model prediction is 28.68

Any help is appreciated as I much prefer lightgbm's implementation of random forest over sklearn's.

Thanks!

![shap_issue](https://user-images.githubusercontent.com/7001021/60271906-c8190c80-98b8-11e9-9d89-ee37d51a31ea.png)

**Comment by slundberg** [comment]
There is a transform between the margin of the model and the output of predict so run `model.predict(X.iloc[0:1], raw_score=True)` and you get the right answer.

**Comment by didericksen** [comment]
Okay thanks for that information. From what I can tell, the raw score is just the model prediction multiplied by the number of trees (500 in this example). When I divide both explainer.expected_value and shap_values by the number of trees, the model outputs match exactly and I get shap values and a force_plot that are similar to the shap values and force_plot when using sklearn.ensemble.RandomForestRegressor. I guess my question then is what advantage is there in having shap values represented in the untransformed space for lightgbm random forest?

**Comment by slundberg** [comment]
Interesting. It is really driven by the choice of LightGBM to encode their RF trees that way. It is a bit odd that they would use a raw_score that is not averaged and then only average the trees as a post-processing step. We could get back to the more expected behavior by adding a special case for LightGBM RF models that divides all leaf values by the number of trees. I'll leave a todo tag here in case anyone wants to tackle that.

**Comment by didericksen** [comment]
Oh I see. Okay thanks again for your help and all your work on shap!

**Comment by jatiniitr** [comment]
I also faced the same issue of sum(Shap values) not matching the predict_proba() output. However, dividing by some constant does not solve the problem in my case. 

Its the logit transformation that did the job; Predict_proba = 1/(1+e^(-raw score)) + constant  

Now I am worried that shap values are getting learned from raw_scores which is not a linear transformation of Predict_proba.


params_lgbm={'num_leaves': 32,
  'max_depth': 4,
  'min_data_in_leaf': 4000,
  'learning_rate': 0.05,
  'n_estimators': 1000,
  'boosting_type': 'gbdt',
  'objective': 'binary',
  'random_state': 501,
  'colsample_bytree': 0.5,
  'subsample': 0.5,
  'reg_alpha': 8,
  'reg_lambda': 12,
  'verbose': -1,
  'CV_n_estimators': 417}

preds=[0.84830326 0.78405538 0.85555192] 
raw_scores = [1.72135477 1.28945767 1.77882661]
sum(shap_values) = [1.31078391, 0.87888681, 1.36825575]

**Comment by ikvision** [comment]
@jatiniitr I am facing the same issue where for most examples  sum(shap_values) do not match preds or raw_scores for LGBM binary classifier.
Any solution?

**Comment by SanGoku95** [comment]
Facing the same issue with LGBM binary classifier
